homeassistant/bootstrap.py

```diff
@@ -336,22 +423,56 @@ def from_config_file(config_path: str,
     hass.config.config_dir = config_dir
```

Comment:
All these things should happen inside the async version. The threaded version should only be a simple wrapper around the async version.